### PR TITLE
CCITCARBON-309 make env variables available during Orchestrate and execute stage of Teflo run

### PR DESCRIPTION
**Pull request by rujutashinde**
* Added environment variables to be passed to the ansible module
* Modified schema to be able to provide env variables in the orchestrate and execute task blocks in the SDF

### CODE REVIEW

**Review by dno-github — APPROVED** [review]
LGTM

**Review by shay6 — APPROVED** [review]

**Comment by shay6** [comment]
LGTM

File: docs/users/definitions/execute.rst [hunk 1]
```diff
@@ -520,6 +527,15 @@ following setting to False in the teflo.cfg
    Teflo expects the xmls collected to have the **<testsuites>** tag  OR **<testsuite>** as its root tag,
    else it skips those xml files for testrun summary generation
 
+Using environment variables:
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In the below example the environment variables data_dir and uname are made available during
+theplaybook execution
```
Comment (shay6) [inline docs/users/definitions/execute.rst:534]: missing space "theplaybook"

File: tests/functional/test_cli.py [hunk 2]
```diff
@@ -279,6 +279,9 @@ def test_iterate_method(mock_method, runner):
             teflo, ['run', '-s', '../localhost_scenario/scenario_local.yml', '-im', 'by_wrong']
         )
         assert results.exit_code == 2
+
+    @staticmethod
+    @mock.patch.object(Teflo, 'run')
```
Comment (shay6) [inline tests/functional/test_cli.py:284]: this is not part of the PR

Comment (rujutashinde) [reply tests/functional/test_cli.py:284]: yes i corrected this which was missing
